master/cmd/determined-master/root.go

```diff
@@ -55,6 +56,12 @@ func runRoot() error {
 	}
 	log.Infof("master configuration: %s", printableConfig)
 
+	err = os.MkdirAll(config.Cache.CacheDir, fs.ModePerm)
```

Comment:
This is a security issue.  fs.ModePerm is `0777` permissions, which would allow any user on the master node to access the content of model definitions.

Let's use the minimum possible permissions, which would be `0700` (read/write/execute for user only, nothing for group and nothing for world).

Reply:
Makes sense, fixed